### PR TITLE
Implement ARO Classic cluster creation/deletion against a deployed full-service RP instance

### PR DESCRIPTION
**Pull request by tsatam**
### Which issue this PR addresses:

Part of [ARO-5121](https://issues.redhat.com/browse/ARO-5121) and [ARO-5122](https://issues.redhat.com/browse/ARO-5122)

### What this PR does / why we need it:

- Implements a `classic-secrets` target to pull credentials for a full-service RP instance from an Azure storage account. 
- Implements the `bootstrap-azure` target and creates a parallel `clean-bootstrap-azure` target to create the backing resources for an ARO cluster (Azure resource group and virtual network w/ master/worker subnets)
- Implements the `classic-create` and `classic-delete` targets to provision ARO clusters using the deployed full-service RP instance. Invoking users are expected to populate the corresponding coordinates and credentials via environment variables. 

### Test plan for issue:

- [X] Manually invoke the targets (on my machine) and confirm they produce the desired results
- [x] Manually invoked the targets from a clean environment and confirm they produce the desired results

### Is there any documentation that needs to be updated for this PR?

Not yet

### CODE REVIEW

**Review by slintes — COMMENTED** [review]
This worked nice after adding my IP to the `rp-nsg` 🙂

I executed:

- make  classic-secrets
- make  classic-bootstrap-azure
- make  classic-create
- make  classic-conftest (successfully listed nodes)
- make  classic-delete
- make  classic-clean-bootstrap-azure (as far as I can tell everything was cleaned up)

Some comments inline.

File: config_example.sh [hunk 1]
```diff
@@ -27,3 +27,57 @@ export CONTAINER_ENGINE_OPTS='--platform linux/amd64'
 # do expire after some time.
 export OPENSHIFT_CI_TOKEN=''
 
+################################################################################
```
[the remark this answers is not in the document]
Comment (tsatam) [reply config_example.sh:30]: I didn't automatically source the config file anywhere in this. We can instrument that directly in a CI job or in the Make targets themselves, I think either option is fine.

File: classic/bootstrap.sh [hunk 2]
```diff
@@ -2,19 +2,59 @@
...
+    if [[ -z "${AZURE_CLUSTER_RESOURCE_GROUP}" ]]; then
+        echo ">> AZURE_CLUSTER_RESOURCE_GROUP is not set"
+        exit 1
+    fi
+
+    echo "Creating resource group ${AZURE_CLUSTER_RESOURCE_GROUP}"
+    az group create \
+        --subscription ${AZURE_SUBSCRIPTION_ID} \
+        --resource-group ${AZURE_CLUSTER_RESOURCE_GROUP} \
+        --location eastus
```
Comment (hlipsig) [inline classic/bootstrap.sh:21]: Given the known capacity issues here until possibly November, should we choose a different location?

**Review by hlipsig — APPROVED** [review]
Suggest change location, but that's non-blocking.

**Pull request merged**